src/sounds.h

```diff
@@ -169,5 +193,100 @@ template<>
+    //
+    // We are in control of reality, and only really care about the perceived sound of creatures at explicit points in time at explicit distances which we control, so we use sound pressure level.
+    short volume;
+
+    // What is the position of the sound source?
+    tripoint origin;
+
+    // What enum sound category is this?
+    sounds::sound_t category;
+
+    // String description of the sound.
+    std::string description;
+
+    // Is this sound from movement?
+    bool movement_noise = false;
+
+    // If all three froms are false, its ambient noise.
+    // Did the player make this noise?
+    bool from_player = false;
+
+    // Did a monster make this noise?
+    bool from_monster = false;
+
+    // Did an NPC make this noise?
+    bool from_npc = false;
+
+    //This stuff is for selecting actual sfx to play through an audio device in THE REAL WORLD. (spooky)
+    std::string id;
+    std::string variant;
+
+    faction_id faction = faction_id( "no_faction" );
+    mfaction_str_id monfaction = mfaction_str_id( "" );
+};
 
+// The dB loss for moving to a new distance, in 100ths of dB. Nominal calc is 100 * (20 * Log10( dist / ( dist - 1 )))
+// dist_vol_loss[2] provides the dB loss moving from 1m to 2m
+// dist_vol_loss[5] provides the dB loss moving from 4m to 5m, etc.
+// Technically we should not have a value for going from 0m to 1m, but we might ask for it with how sound flood filling is handled around corners etc.
+// Mathmatically it should theoretically be somewhere around 20dB, though the rules for calcing sound pressure break down at very small distances
+// And is more of a neat pressure calc or L'Hopital's shenanagins. Just take 15dB.
+//
+// Store this so we dont have to calc distance loss every time we floodfill a tile for sound.
+// These values will be used very frequently, probably a couple hundred times per sound cast for anything but very quiet sounds.
+// Doing the calc out every time for those would bog things down.
+// With this, we should be able to do everything with addition/subtraction.
+// Distance 1 only happens at the source of a sound, i.e., the reference volume.
+//
+// Is this horrific? Yes.
+// Its abjectly heinous, and an afront in the eyes of any diety that would stoop itself to such a point as to cast its gaze upon this wretched datum.
+// Do I care? Yes, it brings me great shame.
+const std::array<short, 122> dist_vol_loss = { 0, 1500, 602, 352, 250, 194, 158, 134, 116, 102, 92, 83, 76, 70, 64, 60, 56, 53, 50, 47, 45, 42, 40, 39, 37, 35, 34, 33, 32, 30, 29, 28, 28, 27, 26, 25, 24, 24, 23, 23, 22, 21, 21, 20, 20, 20, 19, 19, 18, 18, 18, 17, 17, 17, 16, 16, 16, 15, 15, 15, 15, 14, 14, 14, 14, 13, 13, 13, 13, 13, 12, 12, 12, 12, 12, 12, 12, 11, 11, 11, 11, 11, 11, 11, 10, 10, 10, 10, 10, 10, 10, 10, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 7, 7, 7, 7 };
```

Comment:
```suggestion
constexpr auto dist_vol_loss = std::array<short, 122>{ 0, 1500, 602, 352, 250, 194, 158, 134, 116, 102, 92, 83, 76, 70, 64, 60, 56, 53, 50, 47, 45, 42, 40, 39, 37, 35, 34, 33, 32, 30, 29, 28, 28, 27, 26, 25, 24, 24, 23, 23, 22, 21, 21, 20, 20, 20, 19, 19, 18, 18, 18, 17, 17, 17, 16, 16, 16, 15, 15, 15, 15, 14, 14, 14, 14, 13, 13, 13, 13, 13, 12, 12, 12, 12, 12, 12, 12, 11, 11, 11, 11, 11, 11, 11, 10, 10, 10, 10, 10, 10, 10, 10, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 7, 7, 7, 7 };
```
[nitpick]
1. pre-caclculating known amount of values for performance win is a good thing.
2. `constexpr`